### DISCUSSION TITLE
How to read array?

### DISCUSSION THREAD

**Discussion by RhythmLunatic** [vfsfitvnm/frida-il2cpp-bridge#203] [Q&A]
I want to read `System.String[][] arrAVGContent; // 0x120` in class `AVGController`. So I hooked function `System.Void InitializeInternal(System.IO.Stream ms)` in that class, which I think initializes it. (I can't tell for sure, my decompiler failed to decompile).

```ts
var AVGController_InitializeInternal = img.class("AVGController").method("InitializeInternal");
AVGController_InitializeInternal.implementation = function(value: Il2Cpp.Object) : void { 
	

	this.method("InitializeInternal").invoke(value)

	let obj:Il2Cpp.Array<Il2Cpp.Array> = this.field("arrAVGContent").value as Il2Cpp.Array<Il2Cpp.Array> ;
	var l = obj.length;
	for (var i=0;i < l; i++)
	{
		var aa:Il2Cpp.Array = obj.get(i);
		var ll = aa.length;
		for (var j = 0; j < ll; j++)
		{
			console.log(aa.get(j).toString())
		}
	}
}
```
obj is saying access violation if I access it. Maybe I'm casting it wrong. If I cast it like this it also says access violation.
```
let obj:Il2Cpp.Object = this.field("arrAVGContent").value as Il2Cpp.Object ;
console.log(obj.toString());
```

**Comment by vfsfitvnm** [comment]
Casting is not a problem, it's just a TypeScript thing and it doesn't change the actual object type (Howecer, `System.String[][]` corresponds to `Il2Cpp.Array<Il2Cpp.Array<Il2Cpp.String>>`).

Unfortunately I can't help you further, an access violation error is a bad sign and it may depend on factors that are still unknown to me.

Your code looks good to me (except for the formatting :P), so you have to track down that error to a specific line of code or a specific array element.

Well, unless you are trying to access a `NULL` pointer.

**Reply by RhythmLunatic** [reply]
I think the array is not actually used and is never created. But at least I know it wasn't a problem with my code.
